contracts/Org.sol

```diff
@@ -54,40 +58,67 @@ contract Org is Administratable {
 
   /**
    * @notice Create Organization Claim
+   * @param  claimId UUID representing this claim
    * @param  fName First name of Administrator
    * @param  lName Last name of Administrator
    * @param  eMail Email contact for Organization Administrator.
    * @param  orgAdminWalletAddress Wallet address of Organization's Administrator.
    */
   function claimRequest(
+    string memory claimId,
     string memory fName,
     string memory lName,
     string memory eMail,
     address orgAdminWalletAddress
   ) public {
+    require(
+      pendingClaims[claimId].desiredWallet == address(0),
+      "Org: Pending Claim with Id already exists"
+    );
+
     Claim memory newClaim = Claim({
       firstName: fName,
       lastName: lName,
       eMail: eMail,
-      desiredWallet: orgAdminWalletAddress,
-      filesSubmitted: true
+      desiredWallet: orgAdminWalletAddress
     });
-    emit ClaimCreated(newClaim);
-    claims.push(newClaim);
+
+    emit ClaimCreated(claimId, newClaim);
+    pendingClaims[claimId] = newClaim;
   }
 
   /**
    * @notice Approving Organization Claim
-   * @param  index Index value of Claim.
+   * @param claimId UUID of the claim being approved
    * @param adminContractAddress Contract Address for Endaoment Admin
    */
-  function approveClaim(uint256 index, address adminContractAddress)
+  function approveClaim(string calldata claimId, address adminContractAddress)
     public
     onlyAdminOrRole(adminContractAddress, IEndaomentAdmin.Role.REVIEWER)
   {
-    Claim storage claim = claims[index];
-    emit ClaimApproved(claim);
-    orgWallet = claim.desiredWallet;
+    Claim storage claim = pendingClaims[claimId];
```

Comment:
Is there a case where overwriting an already approved claim (`activeClaim`) should be prevented? Just thought I'd ask.

Reply:
Also, would it make any sense to clear the pendingClaims entirely when a claim is approved?

Reply:
dont think we should clear pending claims after an approval. It's possible we'll approve the wrong one and wouldn't want to make an org resubmit the claim.

Reply:
Per comment above, if we don't clear the claim, it lives in Ethereum's state forever, which would make us bad blockchain citizens. From a selfish perspective here, deleting the state from `pendingClaims` would reduce the gas cost to approve.

Reply:
I think we can resolve this discussion, or are there any outstanding issues here?